pymodbus/framer/rtu_framer.py

```diff
@@ -53,12 +53,12 @@ class ModbusRtuFramer(ModbusFramer):
 
     method = "rtu"
 
-    def __init__(self, decoder, client=None):
+    def __init__(self, decoder, **kwargs):
```

Comment:
It does not take any parameter, just client=

it seems you are preparing the code to somehow be overwritten from the outside, but this code is internal.

Reply:
see comment above

Reply:
Now, I remember my problem: [ModbusServerRequestHandler is instantiating the framer](https://github.com/pymodbus-dev/pymodbus/blob/131d30391efd9c64f1fec857ae0372746374e6d2/pymodbus/server/async_io.py#L64C1-L67C14), which means that I have to add transport to each framer constructor even if the framer does not care about transports. How should I handle that?

Reply:
Problem is that framer is supposed to be independent of the physical transport, but you want to interconnect framer and transport.

One way could be to let the transport layer set a uid parameter, that way we would avoid tcp specific code in the framers.